project/Dependencies.scala

```diff
@@ -16,61 +16,79 @@
 
 import sbt._
 
+object Version {
+  val geotools    = "21.2"
+  val spire       = "0.13.0"
+  val accumulo    = "1.9.3"
+  val cassandra   = "3.7.2"
+  val hbase       = "2.2.0"
+  val geomesa     = "2.3.1"
+  val geowave     = "0.9.3"
+  val hadoop      = "2.8.5"
+  val spark       = "2.4.4"
+  val gdal        = "2.4.0"
+  val gdalWarp    = "33.61199eb"
+
+  val previousVersion = "3.0.0"
+}
+import sbt.Keys._
+
 object Dependencies {
+  private def ver(for211: String, for212: String) = Def.setting {
+    CrossVersion.partialVersion(scalaVersion.value) match {
+      case Some((2, 11)) => for211
+      case Some((2, 12)) => for212
+      case _ => sys.error("not good")
+    }
+  }
+
+  def monocle(module: String) = Def.setting {
+    "com.github.julien-truffaut" %% s"monocle-$module" % ver("1.5.1-cats", "2.0.0").value
```

Comment:
Hm I think RasterFrames are using the syntax like this.

Reply:
That's where I lifted this approach but I've certainly seen it in other projects. What do you think? I'm in favor of collapsing the version numbers to here so there is a minimum amount of indirection in the build overall.

Reply:
LGTM 👍